src/Servers/Kestrel/test/InMemory.FunctionalTests/HttpsTests.cs

```diff
@@ -402,15 +402,15 @@ public async Task DevCertWithInvalidPrivateKeyProducesCustomWarning()
                 using (var sslStream = new SslStream(connection.Stream, true, (sender, certificate, chain, errors) => true))
                 {
                     // SslProtocols.Tls is TLS 1.0 which isn't supported by Kestrel by default.
-                    await Assert.ThrowsAsync<IOException>(() =>
+                    await Assert.ThrowsAnyAsync<Exception>(() =>
                         sslStream.AuthenticateAsClientAsync("127.0.0.1", clientCertificates: null,
                             enabledSslProtocols: SslProtocols.Tls,
                             checkCertificateRevocation: false));
                 }
             }
 
             await loggerProvider.FilterLogger.LogTcs.Task.DefaultTimeout();
-            Assert.Equal(3, loggerProvider.FilterLogger.LastEventId);
+            Assert.Equal(1, loggerProvider.FilterLogger.LastEventId);
             Assert.Equal(LogLevel.Error, loggerProvider.FilterLogger.LastLogLevel);
```

Comment:
@Tratcher @javiercn @halter73 I think all of you have played in this area and I could use some help. This test (`DevCertWithInvalidPrivateKeyProducesCustomWarning`) no longer exists in later branches; it was removed as part of 8e1e81ae78e876eecad1b816093f0519682e205d but that relates to product changes that were never backported to 3.1.x. Would like to get it working here or throw the test away.

Right now, the test fails on macOS on the first line (413) because the `LastEventId` actually is `3`. On Linux (which was the platform that caused me to change line 413), the test fails on the second line because the `LastLogLevel` is actually `Debug`. The change here discrepancy to the upgrade to Ubuntu 20.04; the test was working consistently between Ubuntu 18.04 and macOS before the upgrade.

I think my choices are to delete the test,
```suggestion
```
,
```suggestion
            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
            {
                Assert.Equal(1, loggerProvider.FilterLogger.LastEventId);
                Assert.Equal(LogLevel.Debug, loggerProvider.FilterLogger.LastLogLevel);
            } else {
                Assert.Equal(3, loggerProvider.FilterLogger.LastEventId);
                Assert.Equal(LogLevel.Error, loggerProvider.FilterLogger.LastLogLevel);
            }
```
, or perhaps find a way to make `HttpsConnectionFilterLogger` ignore `Debug` events after it's seen an `Error`.

Thoughts❔

Reply:
> , or perhaps find a way to make `HttpsConnectionFilterLogger` ignore `Debug` events after it's seen an `Error`.

This could make sense if the issue is just that a debug log is no occurring after the error. What are the full logs for the failing and passing versions of this tests? This is using `LoggedTest`, so we should have these, right?

Reply:
I'm not seeing anything other than the stack tract that just points to the Assert.Equal failure, unfortunately. There are test .xml/.html files uploaded, but they don't contain any extra info

Reply:
Not sure where the macOS test logs got lost but https://dev.azure.com/dnceng-public/_apis/resources/Containers/28570718/Linux_Test_Logs?itemPath=Linux_Test_Logs%2FInMemory.FunctionalTests%2Fnetcoreapp3.1%2FHttpsTests%2FDevCertWithInvalidPrivateKeyProducesCustomWarning.log at least shows what happened on Ubuntu 20.04:

``` text
[0.233s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HMKM2A62STN9" accepted.
[0.823s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HMKM2A62STN9" started.
[0.825s] [Microsoft.AspNetCore.Server.Kestrel.Https.Internal.HttpsConnectionMiddleware] [Debug] Failed to authenticate HTTPS connection.
System.IO.IOException: Authentication failed because the remote party has closed the transport stream.
   at System.Net.Security.SslStream.StartReadFrame(Byte[] buffer, Int32 readBytes, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslStream.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslStream.ForceAuthentication(Boolean receiveFirst, Byte[] buffer, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslStream.ProcessAuthentication(LazyAsyncResult lazyResult, CancellationToken cancellationToken)
   at System.Net.Security.SslStream.BeginAuthenticateAsServer(SslServerAuthenticationOptions sslServerAuthenticationOptions, CancellationToken cancellationToken, AsyncCallback asyncCallback, Object asyncState)
   at System.Net.Security.SslStream.<>c.<AuthenticateAsServerAsync>b__69_0(SslServerAuthenticationOptions arg1, CancellationToken arg2, AsyncCallback callback, Object state)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncImpl[TArg1,TArg2](Func`5 beginMethod, Func`2 endFunction, Action`1 endAction, TArg1 arg1, TArg2 arg2, Object state, TaskCreationOptions creationOptions)
   at System.Threading.Tasks.TaskFactory.FromAsync[TArg1,TArg2](Func`5 beginMethod, Action`1 endMethod, TArg1 arg1, TArg2 arg2, Object state, TaskCreationOptions creationOptions)
   at System.Threading.Tasks.TaskFactory.FromAsync[TArg1,TArg2](Func`5 beginMethod, Action`1 endMethod, TArg1 arg1, TArg2 arg2, Object state)
   at System.Net.Security.SslStream.AuthenticateAsServerAsync(SslServerAuthenticationOptions sslServerAuthenticationOptions, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Https.Internal.HttpsConnectionMiddleware.InnerOnConnectionAsync(ConnectionContext context) in /_/src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs:line 204
[0.838s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HMKM2A62STN9" stopped.
[1.334s] [TestLifetime] [Information] Finished test DevCertWithInvalidPrivateKeyProducesCustomWarning in 1.3332518s
```

Reply:
If we threw the test away in later versions, I am ok with removing it here to unblock things. We switched most of our coverage to CTI.

Reply:
done as you suggest @javiercn